pkg/pillar/types/sriovtypes.go

```diff
@@ -3,6 +3,141 @@
 
 package types
 
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+// constants for Linux paths for devices
+const (
+	NicLinuxPath     = "/sys/class/net/"
+	NumVfsDevicePath = "/device/sriov_numvfs"
+	VfCountFieldName = "sriov-vf-count"
+	MaxVfCount       = 255
+)
+
+// CreateVF creates Virtual Functions of given count for given Physical Function
+func CreateVF(device string, vfCount uint8) error {
+	name := filepath.Join(NicLinuxPath, device, NumVfsDevicePath)
+	f, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write([]byte(strconv.Itoa(int(vfCount))))
+	if err1 := f.Sync(); err1 != nil && err == nil {
+		err = err1
+	}
+	if err1 := f.Close(); err1 != nil && err == nil {
+		err = err1
+	}
+	return err
+}
+
+// GetVfIfaceName returns formatted VF name
+func GetVfIfaceName(index uint8, ifname string) string {
+	return fmt.Sprintf("%dvf%s", index, ifname)
```

Comment:
What about the opposite order of args, i.e. `eth0vf0` instead of `0vfeth0`? Seems more intuitive to put the parent first and the child next.

Reply:
+1

Reply:
done